[3.0.1-rc1] Stuck at Still creating...

My config is below, I can't seem to get past "Still creating..." with agent set to both 1 or 0. I tried it with the root user and token user described in the install guide as well.  I don't see anything happen in the UI or any requests being made. I do see requests going out and getting 200 OK with: 
```
provider "proxmox" {
  pm_api_url = "${var.proxmoxURL}"
  pm_user = "root@pam"
  pm_password = "<password>"
  pm_log_enable = true
  pm_log_file   = "terraform-plugin-proxmox.log"
  pm_debug      = true
  pm_log_levels = {
    _default    = "debug"
    _capturelog = ""
  }
}
```

I'm using the following:
```
Proxmox 8.1.4
Terraform 1.7.1
Telmate/proxmox 3.0.1-rc1
```

```
  resource "proxmox_vm_qemu" "anima-cluster" {
  count       = var.clusterSize
  name        = "${var.prefix}${count.index + 1}"
  vmid        = var.vmidStartRange + count.index
  ipconfig0  = "ip=${var.networkID}.${count.index + var.networkIPStartRange}${var.networkCIDR},gw=${var.networkID}.1"
  nameserver = "${var.networkID}.1"
  searchdomain = "home.arpa"

  desc        = var.mdClusterDescription
  target_node = var.targetNode

  clone = var.cloudTemplate
  full_clone = true
  onboot = true

  cpu = "host"
  cores   = var.cores
  sockets = var.sockets
  memory  = var.memory
  agent = var.qemuAgent

  disks {
    scsi {
      scsi0 {
        disk {
          storage = var.storageDevice
          size = var.diskSize
          emulatessd = var.ssd
          discard = true
        }
      }
    }
  }

  network {
    bridge    = "vmbr0"
    model     = "virtio"
    firewall  = false
    link_down = false
  }

  os_type   = "cloud-init"
  ciuser = var.username
  cipassword = var.password
  sshkeys = var.sshKeys
}
```

```
proxmoxURL = "https://<domain>:8006/api2/json"
pmUser = "terraform-prov@pve!anima-cluster"
pmTokenSecret = "<token>"
cloudTemplate = "eterna:mantic"
prefix = "anima-"
clusterSize = 6
cores = 4 
sockets = 1
memory = 8192
qemuAgent = 0
targetNode = "pve"
networkID = "10.0.0"
networkCIDR = "/24"
networkIPStartRange = 9
vmidStartRange = 200
storageDevice = "eterna"
diskSize = 100
ssd = true
username = "<username>"
password = "<password>"
sshKeys = <<-EOT
ssh-ed25519 ...
EOT
mdClusterDescription = <<-EOF
# Home Lab Cluster
EOF
```

Mine config looked almost the same before version 3.0.1 when I was using 2.9.3 with Proxmox 7.1. Upgrade to 8.1.3 and got some issues when running terraform with the provider version 2.9.3.. so the next plan was change provider to this one 3.0.1. That gave me resolved issues with the code. I don't remember but there were some issues with string in network rather int and so on. Can't remember. Irrelevant because 3.0.1 fixed that. But.. I stuck on your problem which you are facing "Still creating.." but with small detail related that cloud-init drive is removed during deployment and can't figure it why. And I'm tired of it. Every time a deploy it cloudinit disappear during the process. 

Repo is here, I don't wanna put all content of those files [here](https://github.com/sonic-networks/terraform/tree/master/proxmox/sonic) 

The most important is the null_resource for creating config files

``
null_resource.cloud_init_config_files[0]: Creating...
2024-02-05T18:43:33.699+0100 [INFO]  Starting apply for null_resource.cloud_init_config_files[0]
null_resource.cloud_init_network-config_files[0]: Creating...
2024-02-05T18:43:33.699+0100 [INFO]  Starting apply for null_resource.cloud_init_network-config_files[0]
2024-02-05T18:43:33.699+0100 [DEBUG] null_resource.cloud_init_config_files[0]: applying the planned Create change
2024-02-05T18:43:33.699+0100 [DEBUG] null_resource.cloud_init_network-config_files[0]: applying the planned Create change
null_resource.cloud_init_network-config_files[0]: Provisioning with 'file'...
null_resource.cloud_init_config_files[0]: Provisioning with 'file'...
2024-02-05T18:43:33.701+0100 [DEBUG] Connecting to 10.0.1.1:22 for SSH
2024-02-05T18:43:33.701+0100 [DEBUG] Connecting to 10.0.1.1:22 for SSH
2024-02-05T18:43:33.705+0100 [DEBUG] Connection established. Handshaking for user root
2024-02-05T18:43:33.705+0100 [DEBUG] Connection established. Handshaking for user root
2024-02-05T18:43:33.784+0100 [DEBUG] Telling SSH config to forward to agent
2024-02-05T18:43:33.784+0100 [DEBUG] Setting up a session to request agent forwarding
2024-02-05T18:43:33.789+0100 [DEBUG] Telling SSH config to forward to agent
2024-02-05T18:43:33.789+0100 [DEBUG] Setting up a session to request agent forwarding
2024-02-05T18:43:33.833+0100 [INFO]  agent forwarding enabled
2024-02-05T18:43:33.833+0100 [DEBUG] starting ssh KeepAlives
2024-02-05T18:43:33.833+0100 [DEBUG] opening new ssh session
2024-02-05T18:43:33.841+0100 [INFO]  agent forwarding enabled
2024-02-05T18:43:33.841+0100 [DEBUG] starting ssh KeepAlives
2024-02-05T18:43:33.841+0100 [DEBUG] Starting remote scp process:  'scp' -vt /var/lib/vz/snippets
2024-02-05T18:43:33.841+0100 [DEBUG] opening new ssh session
2024-02-05T18:43:33.844+0100 [DEBUG] Started SCP session, beginning transfers...
2024-02-05T18:43:33.844+0100 [DEBUG] Beginning file upload...
2024-02-05T18:43:33.844+0100 [DEBUG] Starting remote scp process:  'scp' -vt /var/lib/vz/snippets
2024-02-05T18:43:33.848+0100 [DEBUG] SCP session complete, closing stdin pipe.
2024-02-05T18:43:33.848+0100 [DEBUG] Waiting for SSH session to complete.
2024-02-05T18:43:33.853+0100 [DEBUG] Started SCP session, beginning transfers...
2024-02-05T18:43:33.853+0100 [DEBUG] Beginning file upload...
2024-02-05T18:43:33.853+0100 [ERROR] scp stderr: "Sink: C0644 1899 user-data_vm-srv-app-1.yaml\nscp: debug1: fd 0 clearing O_NONBLOCK\r\n"
null_resource.cloud_init_config_files[0]: Creation complete after 0s [id=8398213386398904967]
2024-02-05T18:43:33.856+0100 [DEBUG] SCP session complete, closing stdin pipe.
2024-02-05T18:43:33.856+0100 [DEBUG] Waiting for SSH session to complete.
2024-02-05T18:43:33.858+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-02-05T18:43:33.861+0100 [ERROR] scp stderr: "Sink: C0644 306 network-config_vm-srv-app-1.yaml\nscp: debug1: fd 0 clearing O_NONBLOCK\r\n"
null_resource.cloud_init_network-config_files[0]: Creation complete after 0s [id=4576244245995302674]
2024-02-05T18:43:33.866+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-02-05T18:43:33.866+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-05T18:43:33.867+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/null/3.2.2/darwin_arm64/terraform-provider-null_v3.2.2_x5 pid=36107
2024-02-05T18:43:33.867+0100 [DEBUG] provider: plugin exited
2024-02-05T18:43:33.934+0100 [WARN]  Provider "registry.terraform.io/telmate/proxmox" produced an invalid plan for proxmox_vm_qemu.cloudinit["srv-app-1"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .oncreate: planned value cty.False for a non-computed attribute
      - .force_create: planned value cty.False for a non-computed attribute
      - .define_connection_info: planned value cty.True for a non-computed attribute
      - .preprovision: planned value cty.True for a non-computed attribute
      - .bios: planned value cty.StringVal("seabios") for a non-computed attribute
      - .onboot: planned value cty.False for a non-computed attribute
      - .tablet: planned value cty.True for a non-computed attribute
      - .cpu: planned value cty.StringVal("host") for a non-computed attribute
      - .vlan: planned value cty.NumberIntVal(-1) for a non-computed attribute
      - .clone_wait: planned value cty.NumberIntVal(10) for a non-computed attribute
      - .additional_wait: planned value cty.NumberIntVal(5) for a non-computed attribute
      - .balloon: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .hotplug: planned value cty.StringVal("network,disk,usb") for a non-computed attribute
      - .kvm: planned value cty.True for a non-computed attribute
      - .vcpus: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .full_clone: planned value cty.True for a non-computed attribute
      - .vm_state: planned value cty.StringVal("running") for a non-computed attribute
      - .guest_agent_ready_timeout: planned value cty.NumberIntVal(100) for a non-computed attribute
      - .automatic_reboot: planned value cty.True for a non-computed attribute
      - .smbios: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
      - .disks[0].scsi[0].scsi0[0].disk[0].backup: planned value cty.True for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].mbps_r_burst: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].format: planned value cty.StringVal("raw") for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_r_burst: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].mbps_r_concurrent: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].mbps_wr_burst: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_wr_burst: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_r_burst_length: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].mbps_wr_concurrent: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_wr_burst_length: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_wr_concurrent: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disks[0].scsi[0].scsi0[0].disk[0].iops_r_concurrent: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network[0].firewall: planned value cty.False for a non-computed attribute
      - .network[0].link_down: planned value cty.False for a non-computed attribute
      - .network[0].tag: planned value cty.NumberIntVal(-1) for a non-computed attribute
proxmox_vm_qemu.cloudinit["srv-app-1"]: Creating...
2024-02-05T18:43:33.934+0100 [INFO]  Starting apply for proxmox_vm_qemu.cloudinit["srv-app-1"]
2024-02-05T18:43:33.939+0100 [DEBUG] proxmox_vm_qemu.cloudinit["srv-app-1"]: applying the planned Create change
2024-02-05T18:43:33.948+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:43:33 [DEBUG] setting computed for "smbios" from ComputedKeys: timestamp=2024-02-05T18:43:33.948+0100
2024-02-05T18:43:33.948+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:43:33 [DEBUG] setting computed for "unused_disk" from ComputedKeys: timestamp=2024-02-05T18:43:33.948+0100
2024-02-05T18:43:33.948+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:43:33 [DEBUG][QemuVmCreate] checking for duplicate name: srv-app-1: timestamp=2024-02-05T18:43:33.948+0100
2024-02-05T18:43:33.965+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:43:33 [DEBUG][QemuVmCreate] cloning VM: timestamp=2024-02-05T18:43:33.965+0100
proxmox_vm_qemu.cloudinit["srv-app-1"]: Still creating... [10s elapsed]
proxmox_vm_qemu.cloudinit["srv-app-1"]: Still creating... [20s elapsed]
2024-02-05T18:44:00.128+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:00 [DEBUG][QemuVmCreate] update VM after clone: timestamp=2024-02-05T18:44:00.127+0100
proxmox_vm_qemu.cloudinit["srv-app-1"]: Still creating... [30s elapsed]
proxmox_vm_qemu.cloudinit["srv-app-1"]: Still creating... [40s elapsed]
2024-02-05T18:44:19.826+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:19 [DEBUG][QemuVmCreate] starting VM: timestamp=2024-02-05T18:44:19.825+0100
2024-02-05T18:44:21.892+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:21 [DEBUG][QemuVmCreate] vm creation done!: timestamp=2024-02-05T18:44:21.892+0100
2024-02-05T18:44:21.956+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:21 [DEBUG] VM status: running: timestamp=2024-02-05T18:44:21.955+0100
2024-02-05T18:44:21.956+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:21 [DEBUG] VM is running, checking the IP: timestamp=2024-02-05T18:44:21.956+0100
2024-02-05T18:44:21.956+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:21 [INFO][initConnInfo] trying to get vm ip address for provisioner: timestamp=2024-02-05T18:44:21.956+0100
2024-02-05T18:44:21.956+0100 [INFO]  provider.terraform-provider-proxmox_v3.0.1-rc1: 2024/02/05 18:44:21 [DEBUG][initConnInfo] retrying for at most  20m0s minutes before giving up: timestamp=2024-02-05T18:44:21.956+0100
``

I don't know how to put that damn cloudinit storage when creating the templated. Some of tutorials like this one https://gist.github.com/chriswayg/b6421dcc69cb3b7e41f2998f1150e1df are importing the disk from `local` 
`qm importdisk 9110 debian-10.0.2-20190721-openstack-amd64.qcow2 local -format qcow2` which is likely impossible because `storage local does not support vm images`
Some of you here doesn't even use `cloudinit_cdrom_storage` and some of you are using it although you don't supposed do that when not using `cicustom` according to tf provider docs. I was using official proxmox docs to create template https://pve.proxmox.com/wiki/Cloud-Init_Support and prepared [ansible](https://github.com/sonic-networks/ansible/tree/master) scripts with `command` module and additional conditional so it can run all over again on already created templates by skipping task (command & shell don't provide idempotency) 
So first of all I'm running this script to prepare new templates. And previous code which I shared with you is deploying VMs based on that templates (the one I declare in `clone` argument). Every time running thes damn script everything goes well until suddenly it maks some weird operations on cloudinit and removes it so I have no cloudinit in the VM.

--- a/proxmox/resource_vm_qemu.go
+++ b/proxmox/resource_vm_qemu.go
@@ -805,7 +805,8 @@ func resourceVmQemuCreate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	pconf := meta.(*providerConfiguration)
 	lock := pmParallelBegin(pconf)
-	// defer lock.unlock()
+	defer lock.unlock()
+
 	client := pconf.Client
 	vmName := d.Get("name").(string)
 	vga := d.Get("vga").(*schema.Set)
@@ -1054,6 +1055,7 @@ func resourceVmQemuUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	pconf := meta.(*providerConfiguration)
 	client := pconf.Client
 	lock := pmParallelBegin(pconf)
+	defer lock.unlock()
 
 	// create a logger for this function
 	logger, _ := CreateSubLogger("resource_vm_update")
@@ -1327,6 +1329,7 @@ func resourceVmQemuRead(ctx context.Context, d *schema.ResourceData, meta interf
 	pconf := meta.(*providerConfiguration)
 	lock := pmParallelBegin(pconf)
 	defer lock.unlock()
+
 	client := pconf.Client
 	// create a logger for this function
 	var diags diag.Diagnostics

